### PR TITLE
Add OIDC support & Flask 2.3+ fix

### PR DESCRIPTION
**Pull request by burritosoftware**
Adds OpenID Connect support and fixes an issue with the server not running on Flask 2.3+ due to the deprecation of `@app.before_first_request`, see https://stackoverflow.com/questions/73570041/flask-deprecated-before-first-request-how-to-update.

Needs Black formatter to be run.

### CODE REVIEW

File: food.py [hunk 1]
```diff
@@ -24,19 +24,23 @@
...
-login.init_app(app)
-
 # Ensure the DB is able to determine migration needs.
 migrate = Migrate(app, db, compare_type=True)
 
 
-@app.before_first_request
+@app.before_request
 def initialize_server():
+    # The following line will remove this handler, making it
```
Comment (oscie57) [inline food.py]: This isn't how room-server handles it. Please make it consistent between the two.

File: thepantry/admin.py [hunk 2]
```diff
@@ -1,66 +1,41 @@
...
 from flask import render_template, send_from_directory, redirect, url_for, flash
-from flask_login import current_user, login_user, login_required, logout_user
-from models import User
-from thepantry.forms import LoginForm, ChangePasswordForm
+from flask_oidc import OpenIDConnect
 
+import config
 
-@app.login_manager.unauthorized_handler
-def unauthorized():
```
Comment (oscie57) [inline thepantry/admin.py:9]: Why was the unauthorised handler removed? How does the application react when an unauthorised user tries to make a request?

File: food.py (hunk 1 above)
Comment (oscie57) [reply food.py]: Resolved in latest commit

**Review by oscie57 — APPROVED** [review]
Looks good

File: thepantry/admin.py (hunk 2 above)
Comment (oscie57) [reply thepantry/admin.py:9]: Understandable, resolved

**Comment by burritosoftware** [comment]
Running black to resolve, then will squash and merge

**Comment by burritosoftware** [comment]
Need check bypass from @noahpistilli, if you could check the black formatted files commit, and then turn off Required statuses/checks in the repo settings, thanks

**Pull request merged**